snakeplane/helper_classes.py

```diff
@@ -151,6 +156,16 @@ def all_inputs_completed(self) -> bool:
                     [connection.completed for connection in connections]
                 ):
                     all_inputs_completed = False
+                    
+            if len(self._state_vars.optional_input_names) > 0:  # test if there are any optional anchors
+                for name in self._state_vars.optional_input_names:  # if there are, iterate through them
+                    if self._state_vars.input_anchors[name]:  # check if a optional connection is connected
+                        connections = self._state_vars.input_anchors[name] # has connected input finished?
```

Comment:
You can move this declaration above the `if` statement and then do `if connections`, to simplify the logic a little.